superset-frontend/src/explore/components/controls/FilterBoxItemControl.jsx

```diff
@@ -60,7 +63,35 @@ export default class FilterBoxItemControl extends React.Component {
     this.props.onChange(this.state);
   }
   onControlChange(attr, value) {
-    this.setState({ [attr]: value }, this.onChange);
+    let typedValue = value;
+    const { column: selectedColumnName, multiple } = this.state;
+    if (value && !multiple && attr === 'defaultValue') {
+      // if single value filter_box,
+      // convert input value string to the column's data type
+      const { datasource } = this.props;
+      const selectedColumn = (
+        datasource.columns.filter(
+          col => col.column_name === selectedColumnName,
+        ) || []
+      ).pop();
+
+      if (selectedColumn && selectedColumn.type) {
+        const type = selectedColumn.type.toUpperCase();
+        try {
+          if (type === 'BOOLEAN') {
+            typedValue = value === 'true';
+          } else if (INTEGRAL_TYPES.includes(type)) {
+            typedValue = parseInt(value, 10);
+          } else if (DECIMAL_TYPES.includes(type)) {
+            typedValue = parseFloat(value);
+          }
+        } catch (ex) {
```

Comment:
when does this fail? I don't think `parseInt` or `parseFloat` throw errors with invalid input. instead they return `NaN`. I think we can remove the try/catch block